samples/server/go-server/api/server/storage/pgsqlstore.go

```diff
@@ -0,0 +1,482 @@
+// Copyright 2017 The Grafeas Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/grafeas/grafeas/samples/server/go-server/api/server/name"
+	pb "github.com/grafeas/grafeas/v1alpha1/proto"
+	"github.com/lib/pq"
+	opspb "google.golang.org/genproto/googleapis/longrunning"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type pgSQLStore struct {
+	*sql.DB
+}
+
+func NewPgSQLStore(config *PgSQLConfig) *pgSQLStore {
+	err := createDatabase(createSourceString(config), config.DbName)
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	db, err := sql.Open("postgres", createSourceStringWithDbName(config))
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+	if db.Ping() != nil {
+		log.Fatal("Database server is not alive")
+	}
+	_, err = db.Exec(createTables)
+	if err != nil {
+		db.Close()
+		log.Fatal(err.Error())
+	}
+	pg := pgSQLStore{}
+	pg.DB = db
+	return &pg
+}
+
+func createDatabase(source, dbName string) error {
+	db, err := sql.Open("postgres", source)
+	if err != nil {
+		return err
+	}
+	defer db.Close()
+	// Check if db exists
+	res, err := db.Exec(
+		fmt.Sprintf("SELECT * FROM pg_catalog.pg_database WHERE datname='%s'", dbName))
+	if err != nil {
+		return err
+	}
+	rowCnt, err := res.RowsAffected()
+	if err != nil {
+		return err
+	}
+	// Create database if it doesn't exist
+	if rowCnt == 0 {
+		_, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s", dbName))
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func createSourceString(config *PgSQLConfig) string {
+	return fmt.Sprintf("postgres://%s:%s@%s/?sslmode=%s", config.User, config.Password, config.Host, config.SSLMode)
+}
+
+func createSourceStringWithDbName(config *PgSQLConfig) string {
+	return fmt.Sprintf("postgres://%s:%s@%s/%s?sslmode=%s", config.User, config.Password, config.Host, config.DbName, config.SSLMode)
+}
+
```

Comment:
Can we also put the methods above here into the config file?

Reply:
Done